From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Fredric Silberberg <fredric@fsilberberg.com>
Date: Mon, 12 May 2014 09:48:02 -0400
Subject: [PATCH 0045/6262] WPILibJ Unit Tests

Added unit testing framework for WPILibJ. Tests should based
on the ExampleTests.java located in
src/test/java/edu/wpi/first/wpilibj, and be located in the proper
package.

Change-Id: I19b73e9e24eb8d990d34ab2716d5314c3dee82ef
---
 wpilibj/wpilibJava/pom.xml                    |  5 ++
 .../edu/wpi/first/wpillibj/ExampleTest.java   | 46 +++++++++++++++++++
 2 files changed, 51 insertions(+)
 create mode 100644 wpilibj/wpilibJava/src/test/java/edu/wpi/first/wpillibj/ExampleTest.java

diff --git a/wpilibj/wpilibJava/pom.xml b/wpilibj/wpilibJava/pom.xml
index 383cb772ac48aa6228517bec444fb6417053d68c..e6f61ae28e95eca0482911ea9193337cfc1df1a2 100644
--- a/wpilibj/wpilibJava/pom.xml
+++ b/wpilibj/wpilibJava/pom.xml
@@ -67,6 +67,11 @@
       <artifactId>NetworkTables</artifactId>
       <version>0.1.0-SNAPSHOT</version>
     </dependency>
+    <dependency>
+    	<groupId>junit</groupId>
+    	<artifactId>junit</artifactId>
+    	<version>4.11</version>
+    </dependency>
   </dependencies>
 
   <build>
diff --git a/wpilibj/wpilibJava/src/test/java/edu/wpi/first/wpillibj/ExampleTest.java b/wpilibj/wpilibJava/src/test/java/edu/wpi/first/wpillibj/ExampleTest.java
new file mode 100644
index 0000000000000000000000000000000000000000..42901580c07b07d02e9f545acad0fb974daaffcd
--- /dev/null
+++ b/wpilibj/wpilibJava/src/test/java/edu/wpi/first/wpillibj/ExampleTest.java
@@ -0,0 +1,46 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) FIRST 2008-2012. All Rights Reserved.                        */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+package edu.wpi.first.wpilibj;
+
+import static org.junit.Assert.*;
+import org.junit.Test;
+
+/**
+ * All test classes should be well-documented as to what you are testing. Any
+ * mocks that you create should be put in the same package as the class you are
+ * mocking. Unit tests MUST not depend on any specific hardware platform, so any
+ * dependencies must be mocked out.
+ * 
+ *
+ * Tests are written in with standard JUnit syntax, and are executed with Junit
+ * 4. Tests should be named the same as the unit under test, with "Test"
+ * appended. So a test on the Jaguar class would be called JaguarTest.
+ * 
+ * @author Fredric Silberberg
+ */
+public class ExampleTest {
+
+	/**
+	 * Individual tests should be documented, and naming should be
+	 * "name of thing (unit/function) being tested" + "Test". So a test ensuring
+	 * that an input called "motor" is null-checked correctly would be called
+	 * motorNullCheckTest.
+	 */
+	@Test
+	public void additionTest() {
+		assertEquals(2 + 2, 4);
+	}
+
+	/**
+	 * Tests that expect exceptions should use standard junit expected syntax
+	 */
+	@Test(expected = Exception.class)
+	public void exceptionTest() throws Exception {
+		throw new Exception("This is expected");
+	}
+}
\ No newline at end of file
